api/utxoCache.ts

```diff
@@ -0,0 +1,134 @@
+import { UtxoOrdinalBundle } from '../types/api/xverse/ordinals';
+import { getAddressUtxoOrdinalBundles, getUtxoOrdinalBundle } from './ordinals';
+
+export type UtxoCacheStruct = {
+  [utxoId: string]: UtxoOrdinalBundle;
+};
+
+type UtxoCacheStorage = {
+  version: number;
+  utxos: UtxoCacheStruct;
+};
+
+export type StorageAdapter = {
+  get(key: string): Promise<string | null> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+  remove(key: string): Promise<void> | void;
+};
+
+type UtxoCacheConfig = {
+  cacheStorageController: StorageAdapter;
+};
+
+export class UtxoCache {
+  private readonly _cacheStorageController: StorageAdapter;
+
+  utxos: UtxoCacheStruct = {};
+
+  VERSION = 1;
+
+  constructor(config: UtxoCacheConfig) {
+    this._cacheStorageController = config.cacheStorageController;
+  }
+
+  private getAddressCacheStorageKey = (address: string) => `utxoCache-${address}`;
+
+  _getCache = async (address: string): Promise<UtxoCacheStorage> => {
+    const cache = await this._cacheStorageController.get(this.getAddressCacheStorageKey(address));
+    if (cache) {
+      return JSON.parse(cache);
+    } else {
+      return {
+        version: -1,
+        utxos: {},
+      };
+    }
```

Comment:
Might be good to wrap this in a try catch just in case

```suggestion
    if (cache) {
      try {
        return JSON.parse(cache);
      } catch(err) {
        // noop
      }
    } 
    return {
      version: -1,
      utxos: {},
    };
```